src/Ledger/TokenAlgebra.lagda

```diff
@@ -1,40 +1,54 @@
 \section{Token algebras}
+\label{sec:token-algebra}
+
 \begin{code}[hide]
 {-# OPTIONS --safe #-}
+
 module Ledger.TokenAlgebra where
 
 open import Ledger.Prelude
 
-open import Algebra using (CommutativeMonoid)
-open import Algebra.Morphism
-open import Data.Integer
-open import Data.Nat.Properties using (+-0-commutativeMonoid)
+open import Agda.Primitive       using() renaming (Set to Type)
+open import Algebra              using (CommutativeMonoid ; Monoid)
+open import Algebra.Morphism     using (module MonoidMorphisms )
+open import Data.Nat.Properties  using (+-0-monoid)
+open import Relation.Binary      using (Rel)
+open import Relation.Unary       using (Pred)
 \end{code}
 
 \begin{figure*}
 \begin{code}
 record TokenAlgebra : Set₁ where
-  field PolicyId : Set
+  field -- PolicyId : Set
         Value-CommutativeMonoid : CommutativeMonoid 0ℓ 0ℓ
 
+  MemoryEstimate : Type
   MemoryEstimate = ℕ
 \end{code}
 \begin{code}[hide]
-  open CommutativeMonoid Value-CommutativeMonoid using (_≈_; ε)
-       renaming (Carrier to Value; refl to reflᵛ; _∙_ to _+ᵛ_) public
+  open CommutativeMonoid Value-CommutativeMonoid public
+    using ( _≈_ ; ε ; monoid )
+    renaming  ( Carrier  to Value
+              ; refl     to reflᵛ
+              ; _∙_      to _+ᵛ_
+              )
+  open MonoidMorphisms (Monoid.rawMonoid monoid) (Monoid.rawMonoid +-0-monoid)
 \end{code}
 \begin{code}
-  field coin      : Value → Coin
-        inject    : Coin → Value
-        policies  : Value → ℙ PolicyId
-        size      : Value → MemoryEstimate
-        _≤ᵗ_      : Value → Value → Set
-
-        property             : coin ∘ inject ≗ id
-        coinIsMonoidMorphism : coin Is Value-CommutativeMonoid -CommutativeMonoid⟶ +-0-commutativeMonoid
+  field  coin      : Value → Coin
+         inject    : Coin → Value
+
+         -- policies  : Value → ℙ PolicyId     -- TODO: determine whether we need a policies field
```

Comment:
Yes, this is actually required. We need to put that into `scriptsNeeded`

Reply:
I'm not sure if by "We need to put that into scriptsNeeded" you mean it should be a field in the `TokenAlgebra` record type or it should just be in `scriptsNeeded`.  I'll assume the former, in which case we also need to re-insert the `PolicyId` field, unless `PolicyId` should be a module parameter.  We went over this already in a previous PR that is now closed, so I can probably dig that up and figure it out...

For now I'll make it a field.  Then (I think) when we use the `TokenAlgebra` type, `PolicyId` is supplied in a transaction structure...  (I've forgotten where we put parameters like that... I'll search around.)

Reply:
`PolicyId` needs to be a protocol parameter, so that we can instantiate it with `ScriptHash`, and then we can union `policies (mint tx)` to all the other things in `scriptsNeeded`